src/ray/raylet/cgroup_setup.cc

```diff
@@ -0,0 +1,80 @@
+// Copyright 2024 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/ray/raylet/cgroup_setup.h"
+
+#include <sys/stat.h>
+
+#include <cerrno>
+#include <cstring>
+#include <fstream>
+#include <string_view>
+#include <vector>
+
+#include "src/ray/util/logging.h"
+
+namespace ray {
+
+namespace {
+// Cgroup path for ray system components.
+constexpr std::string_view kSystemCgroupFolder = "/sys/fs/cgroup/system";
```

Comment:
Do we need to worry about potential naming conflict here?

Reply:
That makes sense, append UUID to the path to avoid conflict.